GitHub_Learning/Feature/Dashboard/Presentation/DashboardPresenter.swift

```diff
@@ -27,6 +27,43 @@ final class DashboardPresenterImpl: BasePresenter<DashboardView> {
     
     func viewDidLoad() {
         view?.setEmail(dashboardCotroller.getEmail())
+        dashboardCotroller.subscribe(self, errorBlock: nil, updateBlock: { _ in self.updateBlock()})
+    }
+    
+    private func makeViewModelFrom(_ cryprocurrency: Cryptocurrency?) -> CryptocurrencyViewModel? {
+        guard let cryprocurrencyUnwraped = cryprocurrency else { return nil }
+        let url = cryprocurrencyUnwraped.imageUrl ?? ""
+        let price = String(cryprocurrencyUnwraped.price ?? 0)
+        var priceChangePercentage = cryprocurrencyUnwraped.priceChange?.toString() ?? "N&N"
+        let symbol = cryprocurrencyUnwraped.symbol?.uppercased() ?? "N&N"
+        let priceChange: PriceChangeDirection
+        
+        priceChangePercentage += " %"
+        if priceChangePercentage.contains("-") {
+            priceChange = .negative
+        } else if priceChangePercentage.contains("N&N") || priceChangePercentage.contains("0.0") {
+            priceChange = .neutral
+            priceChangePercentage.insert(contentsOf: "+ ", at: priceChangePercentage.startIndex)
+        } else {
+            priceChange = .positive
+            priceChangePercentage.insert(contentsOf: "+ ", at: priceChangePercentage.startIndex)
+        }
```

Comment:
Cela tato kontrola bude presnejsi, pokud ji budes provadet primo nad tim doublem a ne az nad stringem.

Reply:
Jo to jsem zkoušel akorát mě nenepadl asi dobrý způsob, tím že je to optional. A takhle jsem tam mohl rovnou dýt default value N&N v případě že je to nil. ale zkusim ještě něco vymyslet.

Reply:
Napadlo mě že bych dal jako default value nějaký smyšlený velký číslo který by vyznačovalo nil a na to pak reagoval při vytváření toho stringu abych ho nastavil jako N&N.

Reply:
třeba -100, jelikož -100% nemůže nikdy nastat.